supermarq-benchmarks/supermarq/qcvv/base_experiment.py

```diff
@@ -546,77 +387,100 @@ def run_on_device(
             repetitions=repetitions,
             **target_options,
         )
-        compiled_circuits = experiment_job.compiled_circuits()
-
-        for k, sample in enumerate(self.samples):
-            sample.job = experiment_job[k]
-            sample.compiled_circuit = compiled_circuits[k]
 
-        return experiment_job
+        return self._results_cls(
+            target=target,
+            experiment=self,
+            job=experiment_job,
+        )
 
     def run_with_simulator(
         self,
         simulator: cirq.Sampler | None = None,
         repetitions: int = 10_000,
-        overwrite: bool = False,
-    ) -> None:
+    ) -> ResultsT:
         """Use the local simulator to sample the circuits and store the resulting probabilities.
 
         Args:
             simulator: A local :class:`~cirq.Sampler` to use. If None then the default
                 :class:`cirq.Simulator` simulator is used. Defaults to None.
             repetitions: The number of shots to sample. Defaults to 10,000.
-            overwrite: Whether to force an experiment run even if there is existing data that would
-                be over written in the process. Defaults to False.
-        """
-        if not overwrite:
-            self._has_raw_data()
 
+        Returns:
+            The experiment results object.
+        """
         if simulator is None:
             simulator = cirq.Simulator(seed=self._rng)
 
+        records = []
         for sample in tqdm(self.samples, desc="Simulating circuits"):
             result = simulator.run(sample.circuit, repetitions=repetitions)
             hist = result.histogram(key=cirq.measurement_key_name(sample.circuit))
-            sample.probabilities = {
+            probabilities = {
                 f"{i:0{self.num_qubits}b}": 0.0 for i in range(2**self.num_qubits)
             }  # Set all probabilities to zero
             for val, count in hist.items():
                 # Add in results from the histogram
-                sample.probabilities[f"{val:0{self.num_qubits}b}"] = count / repetitions
+                probabilities[f"{val:0{self.num_qubits}b}"] = count / repetitions
+            records.append({**sample.data, **probabilities})
+
+        return self._results_cls(
+            target="LocalSimulator",
```

Comment:
nit: wdyt about changing this to something like `local_simulator` instead of using camel case (for consistency with non-local targets, e.g. "ibmq_brisbane_qpu")?

Reply:
Updated `Callable` -> `callable` below too